src/platforms/javascript/common/troubleshooting/index.mdx

```diff
@@ -403,3 +403,85 @@ document.body.addEventListener(
 ```
 
 Remember to pass in `true` as the second parameter to `addEventListener()`. Without it, the event handler won't be called, since it's being added to the event target's ancestor rather than the event target itself, and unlike JavaScript runtime errors, `error` events resulting from load failures don't bubble, and therefore must be captured during the `capture` phase. For more information, see [the W3C spec](https://www.w3.org/TR/2003/NOTE-DOM-Level-3-Events-20031107/events.html#Events-phases).
+
+## Reducing Bundle Size
+
+Sentry ships with code that enables you to <PlatformLink to="/configuration/options/#debug">debug your Sentry configuration</PlatformLink>. While this code can be very useful in development environments, it's not typically necessary to include it in your production bundles where it takes up valuable space. The JavaScript SDK uses special flags in its CommonJS and ESM distributions to facilitate tree shaking of code sections that are used for debugging. Tree shaking is a way to remove unused code in your application during the build process.
+
+<PlatformSection notSupported={["javascript.nextjs"]}>
+To mark any debug code as unused, you must replace debug flags in the Sentry SDK with `false`. We've outlined examples of how to do this in popular toolchains below.
+</PlatformSection>
+
+<PlatformSection notSupported={["javascript.nextjs"]}>
+
+### Tree Shaking Debug Code With webpack
+
+To tree shake Sentry debug code in your webpack bundles, we recommend using webpack's [DefinePlugin](https://webpack.js.org/plugins/define-plugin/). Use it to replace all occurrences of the `__SENTRY_DEBUG__` flag with `false`:
+
+```javascript {filename:webpack.config.js}
+const webpack = require("webpack");
+
+module.exports = {
+  // ... other options
+  plugins: [
+    new webpack.DefinePlugin({
+      __SENTRY_DEBUG__: false,
+    }),
+    // ... other plugins
+  ],
+};
+```
+
+</PlatformSection>
+
+<PlatformSection notSupported={["javascript.nextjs"]}>
+
+### Tree Shaking Debug Code With rollup.js
+
+If you're using rollup.js, we recommend [rollup's replace plugin](https://www.npmjs.com/package/@rollup/plugin-replace) to replace all occurrences of `__SENTRY_DEBUG__` with `false`:
+
+```javascript {filename:rollup.config.js}
+import replace from "@rollup/plugin-replace";
+import { terser } from "rollup-plugin-terser";
+
+export default {
+  // ... other options
+  treeshake: "smallest", // recommended for best tree shaking results
+  plugins: [
+    replace({
+      preventAssignment: false, // not necessary because flags are not assigned to
+      values: {
+        __SENTRY_DEBUG__: false,
+      },
+    }),
```

Comment:
```suggestion
    replace({
       __SENTRY_DEBUG__: false,
    }),
```

`preventAssignment` defaults to `false` (and even if it's `true`, it makes no difference, because as you say, it doesn't apply here).

Reply:
> the plugin spits out some message that the default is gonna change soon

Ah, right, had for gotten about this. I agree that's not ideal, but yeah, let's let them worry about that and keep our instructions simple.